### PR TITLE
Fixing Issue #11120: Updating to use semver.valid() instead of semver.range()

### PR DESCRIPTION
**Pull request by issacnitin**
https://github.com/microsoft/azure-pipelines-tasks/issues/11120

### CODE REVIEW

File: Tasks/UseDotNetV2/models.ts [hunk 1]
```diff
@@ -122,7 +122,7 @@ export class VersionParts {
                 throw tl.loc("OnlyExplicitVersionAllowed", version);
             }
 
-            new semver.Range(version);
+            semver.valid(version);
```
Comment (bishal-pdMSFT) [inline Tasks/UseDotNetV2/models.ts]: what is the difference between Range and valid?

Comment (issacnitin) [reply Tasks/UseDotNetV2/models.ts]: semver.valid allows us to validate the version without creating new object. It's more commonly used and more readable as well.

Comment (bishal-pdMSFT) [reply Tasks/UseDotNetV2/models.ts]: @hiyadav  @vineetmimrot  can you please review this PR

Comment (vineetmimrot) [reply Tasks/UseDotNetV2/models.ts]: semver.Range() allows wild characters while doing version checking like 1.2.x. But in this function we are checking for explicit version number and semver.valid() does that. It doesn't allowed wild-characters.

You can further try out the behavior in https://npm.runkit.com/semver

Comment (vineetmimrot) [reply Tasks/UseDotNetV2/models.ts]: @issacnitin : you should check the return value, if it is null then it is invalid entry o/w true

**Pull request merged**